cpp/src/aztec3/circuits/kernel/private/.test.cpp

```diff
@@ -70,288 +61,256 @@ using aztec3::circuits::apps::test_apps::escrow::deposit;
 
 using DummyComposer = aztec3::utils::DummyComposer;
 
-// using aztec3::circuits::mock::mock_circuit;
 using aztec3::circuits::mock::mock_kernel_circuit;
 
+// A type representing any private circuit function
+// (for now it works for deposit and constructor)
+using private_function = std::function<OptionalPrivateCircuitPublicInputs<NT>(
+    FunctionExecutionContext<aztec3::circuits::kernel::private_kernel::Composer>&,
+    std::array<NT::fr, aztec3::ARGS_LENGTH> const&)>;
+
+// Some helper constants for trees
+constexpr size_t MAX_FUNCTION_LEAVES = 2 << (aztec3::FUNCTION_TREE_HEIGHT - 1);
+const NT::fr EMPTY_FUNCTION_LEAF = FunctionLeafPreimage<NT>{}.hash(); // hash of empty/0 preimage
+const NT::fr EMPTY_CONTRACT_LEAF = NewContractData<NT>{}.hash();      // hash of empty/0 preimage
+const auto& EMPTY_FUNCTION_SIBLINGS = compute_empty_sibling_path<NT, aztec3::FUNCTION_TREE_HEIGHT>(EMPTY_FUNCTION_LEAF);
+const auto& EMPTY_CONTRACT_SIBLINGS = compute_empty_sibling_path<NT, aztec3::CONTRACT_TREE_HEIGHT>(EMPTY_CONTRACT_LEAF);
+
 } // namespace
 
 namespace aztec3::circuits::kernel::private_kernel {
 
-class private_kernel_tests : public ::testing::Test {};
-
-TEST(private_kernel_tests, test_deposit)
+/**
+ * @brief Print some debug info about a composer if in DEBUG_PRINTS mode
+ *
+ * @param composer
+ */
+void debugComposer(Composer const& composer)
 {
-    //***************************************************************************
-    // Some private circuit proof (`deposit`, in this case)
-    //***************************************************************************
-
-    const NT::address escrow_contract_address = 12345;
-    // const NT::fr escrow_contract_leaf_index = 1;
-    const NT::fr escrow_portal_contract_address = 23456;
-
-    const NT::fr msg_sender_private_key = 123456789;
-    const NT::address msg_sender =
-        NT::fr(uint256_t(0x01071e9a23e0f7edULL, 0x5d77b35d1830fa3eULL, 0xc6ba3660bb1f0c0bULL, 0x2ef9f7f09867fd6eULL));
-    const NT::address tx_origin = msg_sender;
+#ifdef DEBUG_PRINTS
+    info("computed witness: ", composer.computed_witness);
+    // info("witness: ", private_kernel_composer.witness);
+    // info("constant variables: ", private_kernel_composer.constant_variables);
+    // info("variables: ", composer.variables);
```

Comment:
Not sure we'll want to print these things anymore. I think I might have left these in here by accident when I was learning about the composer.